flatfs.go

```diff
@@ -624,19 +624,24 @@ func (fs *Datastore) putMany(data map[datastore.Key][]byte) error {
 	}
 
 	// move files to their proper places
-	for fi, op := range files {
-		done, err := fs.doWriteOp(op)
+	for _, pop := range files {
+		done, err := fs.doWriteOp(&op{
+			typ:  opRename,
+			key:  pop.key,
+			tmp:  pop.file.Name(),
+			path: pop.path,
+		})
 		if err != nil {
 			return err
-		} else if done {
-			// signify removed
-			ops[fi] = 2
+		} else if !done {
+			_ = os.Remove(pop.file.Name())
```

Comment:
I've convinced myself this works, but there are two things suboptimal we might want to make better:
* The distinction between `pop.path` and `pop.file.Name()` is not well named - which is previous vs new location?
* We're accessing a method (`.Name()`) after `.Close()`ing files. That's relying on the internal behavior to not clear all state when closing the handle, which an implementation would be within its rights to do.